session/create_consumer.go

```diff
@@ -56,9 +57,18 @@ func (consumer *createConsumer) Consume(requestPtr interface{}) (response interf
 		return responseInternalError, err
 	}
 
-	sessionInstance, err := consumer.sessionCreator.Create(consumer.peerID, request.ProposalId, config, destroyCallback)
+	sessionInstance, err := consumer.sessionCreator.Create(consumer.peerID, request.ProposalId, config)
 	switch err {
 	case nil:
+		if destroyCallback != nil {
+			go func() {
+				<-sessionInstance.Stop
+				if err := destroyCallback(); err != nil {
+					log.Error("Failed to execute destroy callback: ", err)
```

Comment:
a prefix here for the logs would be nice

Reply:
Question is the same as with all callbacks - should we need to return an error at all? What caller can do is just log it - the same can do callback.

Reply:
Added log prefix. 
I think it's better to leave for the caller to decide what to do with the error, log it, handle somehow, or retry action.

Reply:
Think of callback as some kind of event notifier (actually this is exactly it - callback to notify that session is gone), it's up to event handler to handle any errors (log, retry) not the caller. This "return error everythere" approach should not be applied blindly.

Reply:
Removed returned error from this callback.